dfp/decimal/decimal

```diff
@@ -180,6 +180,286 @@ public:
 
 namespace std
 {
+
+  namespace
+  {
+    struct _Save_errno
+    {
+      _Save_errno() : _M_errno(errno) { errno = 0; }
+      ~_Save_errno() { if (errno == 0) errno = _M_errno; }
+      int _M_errno;
+    };
+
+    // calls setlocale(LC_ALL, "C") and returns a string containing the old
+    // locale name. Caller must delete[] the string. Returns NULL on error.
+    const char*
```

Comment:
If this code is C++11-only, then this should return `std::unique_ptr<char[]>` and not require the caller to `delete[]` it. That's not an option in the original GCC code because it has to support C++98 still.

Blindly copying the GCC code is not a good idea, there are a number of constraints that apply to GCC which do not apply here. If we didn't need to support C++98 the GCC code would look different.